structures.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+type Service struct {
+	log Logger
+}
+
+type Base struct {
+	Things []interface{}
```

Comment:
you could also have used `json.RawMessage` here 
https://coderoad.ru/48653941/%D0%92-%D1%87%D0%B5%D0%BC-%D1%81%D0%BC%D1%8B%D1%81%D0%BB-json-Rawmessage